go/common/grpc/policy/watcher.go

```diff
@@ -2,35 +2,83 @@ package policy
 
 import (
 	"context"
+	"time"
+
+	"github.com/cenkalti/backoff/v4"
 
 	"github.com/oasisprotocol/oasis-core/go/common"
 	"github.com/oasisprotocol/oasis-core/go/common/accessctl"
 	"github.com/oasisprotocol/oasis-core/go/common/grpc"
 	"github.com/oasisprotocol/oasis-core/go/common/grpc/policy/api"
-	"github.com/oasisprotocol/oasis-core/go/common/pubsub"
+	"github.com/oasisprotocol/oasis-core/go/common/identity"
+	"github.com/oasisprotocol/oasis-core/go/common/logging"
+	"github.com/oasisprotocol/oasis-core/go/common/node"
+	sentry "github.com/oasisprotocol/oasis-core/go/sentry/api"
+	sentryClient "github.com/oasisprotocol/oasis-core/go/sentry/client"
 )
 
 var _ api.PolicyWatcher = (*policyWatcher)(nil)
 
 type policyWatcher struct {
-	policyNotifier *pubsub.Broker
+	ctx context.Context
+
+	sentryAddrs   []node.TLSAddress
+	id            *identity.Identity
+	sentryClients []*sentryClient.Client
+
+	logger *logging.Logger
 }
 
 func (c *policyWatcher) PolicyUpdated(service grpc.ServiceName, accessPolicies map[common.Namespace]accessctl.Policy) {
-	c.policyNotifier.Broadcast(api.ServicePolicies{Service: service, AccessPolicies: accessPolicies})
-}
+	// Notify the sentry nodes of the new policy.
```

Comment:
Can we make sure that this doesn't block the caller?

Reply:
OK, I'll move it to a goroutine.

Reply:
Make sure to add locks around shared state (e.g., `c.sentryClients`).